### PR TITLE
Handle message acknowledgement

### PR DESCRIPTION
**Pull request by a5anka**
Handle message acknowledgments 

Remove messages from the queue when an acknowledgment is received.

Connect basic.ack frame handle with broker core. A unacked message map is maintained in the AMQ channel to get the matching ack information for a received delivery tag.

N/A

* Handle Message Acknowledgements

N/A

N/A

N/A

N/A

## Automation tests
 - Unit tests 74% 

 - Integration tests - already exists

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: modules/broker-amqp/src/main/java/org/wso2/broker/amqp/codec/AmqpChannel.java [hunk 1]
```diff
@@ -72,9 +91,9 @@ public ShortString consume(ShortString queueName, ShortString consumerTag, boole
                                ChannelHandlerContext ctx) throws BrokerException {
         String tag = consumerTag.toString();
         if (tag.isEmpty()) {
-            tag = UUID.randomUUID().toString();
+            tag = "sgen" + getNextConsumerTag();
```
Comment (Asitha) [inline modules/broker-amqp/src/main/java/org/wso2/broker/amqp/codec/AmqpChannel.java]: what is meant by `sgen`?

File: modules/broker-core/src/main/java/org/wso2/broker/core/ExchangeRegistry.java [hunk 2]
```diff
@@ -120,7 +102,7 @@ void declareExchange(String exchangeName, Exchange.Type type,
 
         static Exchange newInstance(String exchangeName, Exchange.Type type) throws BrokerException {
             Exchange exchange;
-            switch(type) {
+            switch (type) {
```
Comment (Asitha) [inline modules/broker-core/src/main/java/org/wso2/broker/core/ExchangeRegistry.java]: We didn't get a checkstyle issue for this? Hmm...

File: modules/broker-amqp/src/main/java/org/wso2/broker/amqp/codec/AmqpChannel.java (hunk 1 above)
Comment (a5anka) [reply modules/broker-amqp/src/main/java/org/wso2/broker/amqp/codec/AmqpChannel.java]: shortcode for server-generated. Qpid has also used a similar thing

File: modules/broker-core/src/main/java/org/wso2/broker/core/ExchangeRegistry.java (hunk 2 above)
Comment (a5anka) [reply modules/broker-core/src/main/java/org/wso2/broker/core/ExchangeRegistry.java]: Nope

**Pull request merged**